### ISSUE TITLE
Cannot access admin interface using default configuration

### ISSUE THREAD

**Issue by sadpatsfan** [jc21/nginx-proxy-manager#1101]
When I try to access the host on port 81 it just ties out.  If I try on port 80 I get the nginx page telling me I need to set it up.

I have create a docker-compose file following the example, but once the containers are up I am unable to connect on port 81 to configure nginx.  The error.log shows me:

connect() failed (111: Connection refused) while connecting to upstream, client: 127.0.0.1, server: nginxproxymanager, request: "GET /api/ HTTP/1.1", upstream: "http://127.0.0.1:3000/", host: "127.0.0.1:81"

Here is my compose file:
version: "3"
services:
  app:
    image: 'jc21/nginx-proxy-manager:latest'
    restart: always
    ports:
      - '80:80'
      - '443:443'
      - '81:81'
    environment:
      DB_MYSQL_HOST: "db"
      DB_MYSQL_PORT: 3306
      DB_MYSQL_USER: "npm"
      DB_MYSQL_PASSWORD: "npm"
      DB_MYSQL_NAME: "npm"
      DISABLE_IPV6: 'true'
    volumes:
      - proxy_data:/data
      - proxy_letsencrypt:/etc/letsencrypt
    depends_on:
      - db
  db:
    image: 'jc21/mariadb-aria:latest'
    restart: always
    environment:
      MYSQL_ROOT_PASSWORD: 'npm'
      MYSQL_DATABASE: 'npm'
      MYSQL_USER: 'npm'
      MYSQL_PASSWORD: 'npm'
    volumes:
      - proxy_mysql:/var/lib/mysql

volumes:
  proxy_data:
  proxy_letsencrypt:
  proxy_mysql:
    driver: local

networks:
  default:
    external:
      name: proxy

What am I missing or doing wrong?  I do not have any other files in the directory, running this on a pretty clean ubuntu instance (on AWS, if that matters...security group allows 80,443 and 81) which currently does not have any other containers running (it will if I can ever get this working!).

**Comment by jc21** [comment]
what's in the docker logs for both containers?

**Comment by sadpatsfan** [comment]
hmm, so this is new, and I just realized that I had not checked these logs since changing the compose file to use volumes instead of files on the host.  This is what I see in the app log now:
[5/18/2021] [12:36:52 AM] [Global   ] › ℹ  info      Generating MySQL db configuration from environment variables
[5/18/2021] [12:36:52 AM] [Global   ] › ℹ  info      Wrote db configuration to config file: ./config/production.json
[5/18/2021] [12:36:54 AM] [Global   ] › ✖  error     select * from `migrations_lock` - ER_UNKNOWN_STORAGE_ENGINE: Unknown storage engine 'InnoDB'

and in the db log:
[i] pre-init.d - processing /scripts/pre-init.d/01_secret-init.sh
[i] mysqld not found, creating....
[i] MySQL directory already present, skipping creation
2021-05-18  0:36:51 0 [Note] /usr/bin/mysqld (mysqld 10.4.15-MariaDB) starting as process 1 ...
2021-05-18  0:36:51 0 [Note] Plugin 'InnoDB' is disabled.
2021-05-18  0:36:51 0 [Note] Plugin 'FEEDBACK' is disabled.
2021-05-18  0:36:51 0 [Note] Server socket created on IP: '::'.
2021-05-18  0:36:51 0 [Warning] 'proxies_priv' entry '@% root@be036e47b0e9' ignored in --skip-name-resolve mode.
2021-05-18  0:36:51 1 [Warning] Failed to load slave replication state from table mysql.gtid_slave_pos: 1286: Unknown storage engine 'InnoDB'
2021-05-18  0:36:51 0 [Note] Reading of all Master_info entries succeeded
2021-05-18  0:36:51 0 [Note] Added new Master_info '' to hash table
2021-05-18  0:36:51 0 [Note] /usr/bin/mysqld: ready for connections.
Version: '10.4.15-MariaDB'  socket: '/run/mysqld/mysqld.sock'  port: 3306  MariaDB Server


I am going to revert back to directories instead of volumes and check the logs again.

**Comment by sadpatsfan** [comment]
ok,  reverted back to this compose:

version: "3"
services:
  app:
    image: 'jc21/nginx-proxy-manager:latest'
    restart: always
    ports:
      - '80:80'
      - '443:443'
      - '81:81'
    environment:
      DB_MYSQL_HOST: "db"
      DB_MYSQL_PORT: 3306
      DB_MYSQL_USER: "npm"
      DB_MYSQL_PASSWORD: "npm"
      DB_MYSQL_NAME: "npm"
      # If you would rather use Sqlite uncomment this
      # and remove all DB_MYSQL_* lines above
#       DB_SQLITE_FILE: "/data/database.sqlite"
      # Uncomment this if IPv6 is not enabled on your host
      DISABLE_IPV6: 'true'
    volumes:
      - ./data:/data
      - ./letsencrypt:/etc/letsencrypt
    depends_on:
      - db
  db:
    image: 'jc21/mariadb-aria:latest'
#    image: mariadb:latest
    restart: always
    environment:
      MYSQL_ROOT_PASSWORD: 'npm'
      MYSQL_DATABASE: 'npm'
      MYSQL_USER: 'npm'
      MYSQL_PASSWORD: 'npm'
    volumes:
      - ./mysql:/var/lib/mysql

#volumes:
#  proxy_data:
#  proxy_letsencrypt:
#  proxy_mysql:
#    driver: local

networks:
  default:
    external:
      name: proxy



I still have the same issue, and see the same error in the error.log file.

The docker app logs now show:
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      Current database version: none
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [initial-schema] Migrating Up...
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [initial-schema] auth Table created
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [initial-schema] user Table created
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [initial-schema] user_permission Table created
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [initial-schema] proxy_host Table created
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [initial-schema] redirection_host Table created
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [initial-schema] dead_host Table created
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [initial-schema] stream Table created
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [initial-schema] access_list Table created
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [initial-schema] certificate Table created
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [initial-schema] access_list_auth Table created
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [initial-schema] audit_log Table created
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [websockets] Migrating Up...
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [websockets] proxy_host Table altered
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [forward_host] Migrating Up...
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [forward_host] proxy_host Table altered
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [http2_support] Migrating Up...
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [http2_support] proxy_host Table altered
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [http2_support] redirection_host Table altered
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [http2_support] dead_host Table altered
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [forward_scheme] Migrating Up...
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [forward_scheme] proxy_host Table altered
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [disabled] Migrating Up...
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [disabled] proxy_host Table altered
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [disabled] redirection_host Table altered
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [disabled] dead_host Table altered
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [disabled] stream Table altered
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [custom_locations] Migrating Up...
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [custom_locations] proxy_host Table altered
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [hsts] Migrating Up...
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [hsts] proxy_host Table altered
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [hsts] redirection_host Table altered
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [hsts] dead_host Table altered
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [settings] Migrating Up...
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [settings] setting Table created
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [access_list_client] Migrating Up...
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [access_list_client] access_list_client Table created
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [access_list_client] access_list Table altered
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [access_list_client_fix] Migrating Up...
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [access_list_client_fix] access_list Table altered
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [pass_auth] Migrating Up...
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [pass_auth] access_list Table altered
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [redirection_scheme] Migrating Up...
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [redirection_scheme] redirection_host Table altered
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [redirection_status_code] Migrating Up...
[5/18/2021] [12:40:42 AM] [Migrate  ] › ℹ  info      [redirection_status_code] redirection_host Table altered
[5/18/2021] [12:40:42 AM] [Setup    ] › ℹ  info      Creating a new JWT key pair...
[5/18/2021] [12:40:47 AM] [Setup    ] › ℹ  info      Wrote JWT key pair to config file: /app/config/production.json
[5/18/2021] [12:40:47 AM] [Setup    ] › ℹ  info      Creating a new user: admin@example.com with password: changeme
[5/18/2021] [12:40:48 AM] [Setup    ] › ℹ  info      Initial admin setup completed
[5/18/2021] [12:40:48 AM] [Setup    ] › ℹ  info      Default settings added
[5/18/2021] [12:40:48 AM] [IP Ranges] › ℹ  info      Fetching IP Ranges from online services...
[5/18/2021] [12:40:48 AM] [IP Ranges] › ℹ  info      Fetching https://ip-ranges.amazonaws.com/ip-ranges.json
[5/18/2021] [12:40:48 AM] [IP Ranges] › ℹ  info      Fetching https://www.cloudflare.com/ips-v4
[5/18/2021] [12:40:48 AM] [IP Ranges] › ℹ  info      Fetching https://www.cloudflare.com/ips-v6
[5/18/2021] [12:40:48 AM] [SSL      ] › ℹ  info      Let's Encrypt Renewal Timer initialized
[5/18/2021] [12:40:48 AM] [SSL      ] › ℹ  info      Renewing SSL certs close to expiry...
[5/18/2021] [12:40:48 AM] [IP Ranges] › ℹ  info      IP Ranges Renewal Timer initialized
[5/18/2021] [12:40:48 AM] [Global   ] › ℹ  info      Backend PID 223 listening on port 3000 ...
[5/18/2021] [12:40:50 AM] [Nginx    ] › ℹ  info      Reloading Nginx
[5/18/2021] [12:40:50 AM] [SSL      ] › ℹ  info      Renew Complete

and the db log shows:
[i] pre-init.d - processing /scripts/pre-init.d/01_secret-init.sh
[i] mysqld not found, creating....
[i] MySQL data directory not found, creating initial DBs
[i] Creating database: npm
[i] with character set: 'utf8' and collation: 'utf8_general_ci'
[i] Creating user: npm with password npm
2021-05-18  0:40:40 0 [Note] /usr/bin/mysqld (mysqld 10.4.15-MariaDB) starting as process 52 ...
2021-05-18  0:40:40 0 [Note] Plugin 'InnoDB' is disabled.
2021-05-18  0:40:40 0 [Note] Plugin 'FEEDBACK' is disabled.
2021-05-18  0:40:40 1 [Warning] 'user' entry '@09e3ca668db6' ignored in --skip-name-resolve mode.
2021-05-18  0:40:40 1 [Warning] 'proxies_priv' entry '@% root@09e3ca668db6' ignored in --skip-name-resolve mode.
2021-05-18  0:40:40 1 [Warning] 'user' entry '@09e3ca668db6' ignored in --skip-name-resolve mode.
2021-05-18  0:40:40 1 [Warning] 'proxies_priv' entry '@% root@09e3ca668db6' ignored in --skip-name-resolve mode.
/scripts/run.sh: ignoring or entrypoint initdb empty /docker-entrypoint-initdb.d/*


MySQL init process done. Ready for start up.

exec /usr/bin/mysqld --user=mysql --console --skip-name-resolve --skip-networking=0
2021-05-18  0:40:40 0 [Note] /usr/bin/mysqld (mysqld 10.4.15-MariaDB) starting as process 1 ...
2021-05-18  0:40:40 0 [Note] Plugin 'InnoDB' is disabled.
2021-05-18  0:40:40 0 [Note] Plugin 'FEEDBACK' is disabled.
2021-05-18  0:40:40 0 [Note] Server socket created on IP: '::'.
2021-05-18  0:40:40 0 [Warning] 'user' entry '@09e3ca668db6' ignored in --skip-name-resolve mode.
2021-05-18  0:40:40 0 [Warning] 'proxies_priv' entry '@% root@09e3ca668db6' ignored in --skip-name-resolve mode.
2021-05-18  0:40:40 0 [Note] Reading of all Master_info entries succeeded
2021-05-18  0:40:40 0 [Note] Added new Master_info '' to hash table
2021-05-18  0:40:40 0 [Note] /usr/bin/mysqld: ready for connections.
Version: '10.4.15-MariaDB'  socket: '/run/mysqld/mysqld.sock'  port: 3306  MariaDB Server


Thanks in advance for your help.

**Comment by jc21** [comment]
Looks like you're good go so not sure why you wouldn't be able to access on port 81 now. try

```bash
docker exec -ti npm curl http://127.0.0.1:81/api/
```
and if that fails
```bash
docker exec -ti npm curl http://127.0.0.1:3000
```
and see what you get

**Comment by sadpatsfan** [comment]
Error: No such container: npm

however,
docker exec -ti proxy_app_1 curl http://127.0.0.1:81/api/
{"status":"OK","version":{"major":2,"minor":9,"revision":2}}

docker exec -ti proxy_app_1 curl http://127.0.0.1:3000
{"status":"OK","version":{"major":2,"minor":9,"revision":2}}

**Comment by sadpatsfan** [comment]
OK, so I have no explanation, but it is working now.  All I did was run docker-compose down, then I deleted the compose file and created a new one, pasted in the exact same config, adding nothing, brought it up and tried it, worked right away.  I literally didn't change a single character in the file (I copied it to my clipboard, and pasted it into a new file,that's it). Very weird.  Sorry for wasting your time and thank you for the help! Hopefully it's smooth sailing from here.
